test/blackbox/common/DDSBlackboxTestsDiscovery.cpp

```diff
@@ -52,4 +61,110 @@ TEST(DDSDiscovery, IgnoreParticipantFlags)
     p3.init();
     EXPECT_TRUE(p1.wait_participant_discovery());
     EXPECT_TRUE(p3.wait_participant_discovery());
-}
\ No newline at end of file
+}
+
+/**
+ * This test checks that adding servers to the Discovery Server list results in discovering those participants.
+ * It does so by:
+ *    1. Creating two servers and one client that is only connected to one server. At this point check discovery
+ *       state.
+ *    2. Then, connect the client to the other server and check discovery again.
+ *    3. Finally connect the two servers by adding one of them to the others list
+ */
+TEST(DDSDiscovery, AddDiscoveryServerToList)
+{
+    using namespace eprosima::fastdds::dds;
+    using namespace eprosima::fastrtps::rtps;
+
+    /* Get random port from the environment */
+    char* value = nullptr;
+    value = std::getenv("W_UNICAST_PORT_RANDOM_NUMBER");
+    if (nullptr == value)
+    {
+        value = &std::string("11811")[0];
+    }
+
+    /* Create first server */
+    PubSubParticipant<HelloWorldType> server_1(0u, 0u, 0u, 0u);
+    // Set participant as server
+    WireProtocolConfigQos server_1_qos;
+    server_1_qos.builtin.discovery_config.discoveryProtocol = DiscoveryProtocol_t::SERVER;
+    // Generate random GUID prefix
+    srand((unsigned)time(NULL));
```

Comment:
This is a small nitpick (and a come-back from a previous review :joy:)

```suggestion
    srand(static_cast<unsigned>(time(nullptr)));
```

More C++ like :stuck_out_tongue_closed_eyes:

Reply:
Didnn't you like the `nullptr` instead of the `NULL`?

Reply:
> Didnn't you like the `nullptr` instead of the `NULL`?

🤦🏼‍♂️ I missed that!